homeassistant/components/media_player/plex.py

```diff
@@ -62,10 +101,13 @@ def config_from_file(filename, config=None):
 
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup the Plex platform."""
-    config = config_from_file(hass.config.path(PLEX_CONFIG_FILE))
-    if len(config):
+    # get config from plex.conf
+    file_config = config_from_file(hass.config.path(PLEX_CONFIG_FILE))
+    config.update(file_config)  # merge configs
```

Comment:
We should not mutate the passed in config. I also think that it's kinda weird, because the config in the file contains ip adresses + tokens ? 

To copy the config, do:

```python
config = dict(config)
```